src/app/authorisation/auth.guard.ts

```diff
@@ -0,0 +1,25 @@
+import { Injectable } from '@angular/core';
+import {
+  ActivatedRouteSnapshot,
+  CanActivate,
+  Router,
+  RouterStateSnapshot,
+  UrlTree,
+} from '@angular/router';
+import { Observable } from 'rxjs';
+import { AuthService } from './auth.service';
+
+@Injectable({
+  providedIn: 'root',
+})
+export class AuthGuard implements CanActivate {
+  constructor(private authService: AuthService, private router: Router) {}
+  canActivate() {
+    if (this.authService.isLoggedIn()) {
+      return true;
+    }
+    alert('You have not logged in!');
+    this.router.navigate(['login']);
+    return false;
```

Comment:
```suggestion
    return this.router.createUrlTree(['login']);;
```
Angular Guards have a standard API for redirection, where they can return a `UrlTree`, which will tell Angular to redirect to somewhere else. UrlTrees should be used insted of router navigation.